use-cases/vehicle-routing-capacity/src/main/java/org/acme/vehiclerouting/domain/VehicleRoutePlan.java

```diff
@@ -0,0 +1,170 @@
+package org.acme.vehiclerouting.domain;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningEntityCollectionProperty;
+import ai.timefold.solver.core.api.domain.solution.PlanningScore;
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.domain.solution.ProblemFactCollectionProperty;
+import ai.timefold.solver.core.api.domain.valuerange.ValueRangeProvider;
+import ai.timefold.solver.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
+import ai.timefold.solver.core.api.solver.SolverStatus;
+
+import org.acme.vehiclerouting.domain.geo.DistanceCalculator;
+import org.acme.vehiclerouting.domain.geo.EuclideanDistanceCalculator;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@PlanningSolution
+public class VehicleRoutePlan {
+
+    private String name;
+
+    @JsonIgnore
+    private List<Location> locations;
+
+    @ProblemFactCollectionProperty
+    private List<Depot> depots;
+
+    @PlanningEntityCollectionProperty
+    private List<Vehicle> vehicles;
+
+    @ProblemFactCollectionProperty
+    @ValueRangeProvider
+    private List<Customer> customers;
+
+    @PlanningScore
+    private HardSoftLongScore score;
+
+    private Location southWestCorner;
+    private Location northEastCorner;
+
+    private SolverStatus solverStatus;
+
+    private String scoreExplanation;
```

Comment:
Interesting and good! This is a step forward, so let's add this indeed, but I do think we need a broader discussion to figure out the best way to do this.

Reply:
This was already in place; but I agree we need to return the data structured; the justification String is just a poor man's solution.
We might have a sub-resource: `/route-plans/UUID/explanation` or even something richer.